cmd/object-handlers.go

```diff
@@ -766,8 +766,10 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		return
 	}
 
+	// Check if bucket encryption is enabled
+	_, encEnabled := globalBucketSSEConfigSys.Get(dstBucket)
 	// This request header needs to be set prior to setting ObjectOptions
-	if globalAutoEncryption && !crypto.SSEC.IsRequested(r.Header) {
+	if globalAutoEncryption || encEnabled && !crypto.SSEC.IsRequested(r.Header) {
```

Comment:
```suggestion
      if (globalAutoEncryption || encEnabled) && !crypto.SSEC.IsRequested(r.Header) {
```

otherwise if globalAutoEncryption is set and ssec headers are also set,SSE-S3 header will be added to request  incorrectly